pkgs/development/libraries/phonon/backends/gstreamer.nix

```diff
@@ -29,6 +29,23 @@ stdenv.mkDerivation rec {
     sha256 = "1wc5p1rqglf0n1avp55s50k7fjdzdrhg0gind15k8796w7nfbhyf";
   };
 
+  # Hardcode paths to useful plugins so the backend doesn't depend
+  # on system paths being set.
+  patches = [ ./gst-plugin-paths.patch ];
+  gstPluginPaths =
+    lib.makeSearchPathOutput "lib" "/lib/gstreamer-1.0"
+    (with gst_all_1; [
+      gstreamer
+      gst-plugins-base
+      gst-plugins-good
+      gst-plugins-ugly
+      gst-plugins-bad
+      gst-libav
+    ]);
+  postPatch = ''
+    substituteInPlace gstreamer/backend.cpp --subst-var gstPluginPaths
```

Comment:
What I mean is, you could also write it like this to eliminate the substitution (particularly so since the patch is a small one):

```
patches = [
  (writeTextFile
    { name = "gst-plugins-paths.patch";
      text = ''
        Index: phonon-gstreamer-4.9.0/gstreamer/backend.cpp
        ===================================================================
        --- phonon-gstreamer-4.9.0.orig/gstreamer/backend.cpp
        +++ phonon-gstreamer-4.9.0/gstreamer/backend.cpp
        @@ -85,6 +85,8 @@ Backend::Backend(QObject *parent, const
                 "--gst-debug-no-color"
             };
         
        +    qputenv("GST_PLUGIN_PATH_1_0", "${gstPluginsPath}");
        +
             int argc = sizeof(args) / sizeof(*args);
             char **argv = const_cast<char**>(args);
             GError *err = 0;
        '';
    })
];
```